Download initial domain db

NeelSarwal opened the pull request:
Think I still need to compare hashes for better security

Commit:
Download initial domain db

clintonb commented on voripos-provision.sh:
```diff
@@ -183,6 +184,22 @@ defaults write com.vori.VoriPOS provisioned_litestreamRegion -string "$litestrea
+# Initial domain database download when not reprovisioning
+if [ "$reprovision" = false ] ; then
+  echo "Downloading initial domain database..."
+  response=$(curl --silent -w "\n%{http_code}" -L -X GET "$VORI_API_ROOT/pos/domain-data" -H "Content-Type: application/json" -H "X-Vori-Voripos-Provision-Version: $VORIPOS_PROVISION_VERSION" "${curlArgs[@]}")
```
This won’t work because the endpoint requires an OAuth token/JWT.

clintonb commented on voripos-provision.sh:
```diff
@@ -6,6 +6,7 @@ set +x
 
 VORIPOS_PROVISION_VERSION=0.10.0
 VORI_API_ROOT="${VORI_API_ROOT:-https://api.vori.com/v1}"
+DOMAIN_FILE_PATH="~/Library/Containers/com.vori.VoriPOS/Data/Library/Application Support/domain"
```
Use $HOME instead of tilde for maximum compatibility. Also, capitalize Domain (even though it doesn’t matter on macOS).

clintonb reviewed (changes requested):
Nice work! I can finish up the token request and get this merged.

clintonb committed:
Many updates

- Updated version to 0.11.0
- Added option to download domain data when re-provisioning
- Fixed cURL issues
- Resolved all ShellCheck warnings

clintonb committed:
fixup! Many updates
